### ISSUE TITLE
Is there any compatibility matrix?

### ISSUE THREAD

**Issue by smiklosovic** [operator-framework/operator-sdk#2452]
Hi,

I am in advance sorry for my uninformed question but ...

I was on operator-sdk version 0.10.0 and I was using it from times of Kubernetes cluster 1.14 to 1.17.1 and all went good.

I have just updated operator to 0.11.0 and again all seems to be good with my version of cluster 1.17.1.

I know that operator-sdk 0.11.0 was "pinned" to use some APIs of Kubernetes 1.14.1 but what does this mean exactly? Imagine a customer who is running a cluster of version 1.14.9. Operator 0.12.0 was "pinned" to 1.15.4 (I think) so it means that once I use operator 0.12.0, it will not be supported on 1.14.9? Does it mean that I have to use operator-sdk of "lower" version?

**Comment by joelanford** [comment]
@smiklosovic

**TL;DR:**
The plan is that we'll list the one Kubernetes version that we build and test against and then link to the client-go compatibility matrix to help users understand what Kubernetes versions can be expected to work (or not work) with each version of Operator SDK.

Check out #2446 for details on our planned approach.

**Longer explanation:**
The version that is in SDK's `go.mod` file is the version we're "pinned" to. That basically means that the set of APIs available to clients is based on the set of APIs that that Kubernetes version supports.

So if you're using operator-sdk v0.10.0, which uses kubernetes 1.13.4 client libraries against a kubernetes 1.17.1 server, there will likely be some things that are supported (e.g. Pods, Services, etc) since their APIs haven't changed, some things that are broken (e.g. https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.16.md#deprecations-and-removals) since they've been removed, and some things you can't take advantage of (e.g. `CustomResourceDefinition` at `v1`) since they were added in a later version than your client supports.

Anytime your client and server versions are not aligned, the answer to the question "is my operator compatible with kubernetes server at version X?" is "it depends".

If you have any follow-ups, feel free to re-open and we can continue the discussion.
